cms/djangoapps/contentstore/views/item.py

```diff
@@ -1230,6 +1230,7 @@ def create_xblock_info(xblock, data=None, metadata=None, include_ancestor_info=F
             'graded': xblock.graded,
             'due_date': get_default_time_display(xblock.due),
             'due': xblock.fields['due'].to_json(xblock.due),
+            'due_num_weeks': xblock.fields['due_num_weeks'].to_json(xblock.due_num_weeks),
```

Comment:
Ah yeah you're right, I think `to_json` is unnecessary here! I was initially trying to replicate the 'due' attribute where `to_json` seems to be necessary since it's a Date object, but due_num_weeks is just an Integer so I also don't think it's needed here.

Reply:
It actually seems like the Integer object needs to be JSON serializable 🤔

Reply:
yup it's a giant dictionary of all the information for the xblock and its children! I get a bunch of "Object of type Integer is not JSON serializable" for the existing tests -- I was guessing it might have to do with JS Integer object vs. int in python?

Reply:
What if you just did `'due_num_weeks': xblock.due_num_weeks`? Similar to the graded attribute. Wonder if the whole xblock.fields thing is necessary at all. Also not sure what data type it is and what happens with the `.to_json` and then the input parameter to that.

Reply:
just doing `due_num_weeks': xblock.due_num_weeks` works now as an 'int' type! It seems like  `xblock.fields['due_num_weeks']` is an Integer object, and if I do `xblock.fields['due_num_weeks'].to_json(xblock.due_num_weeks)` I'm making the Integer object into an `int` type (so I'm unnecessarily doing two steps when I could just do `xblock.due_num_weeks` like you said!)

Reply:
Hmm. Interesting stuff. I wonder how the heck the `to_json` method works since it was taking that int as a parameter. Oh well. Glad we found something more simple.